app/assets/javascripts/discourse/components/desktop-notification-config.js.es6

```diff
@@ -0,0 +1,61 @@
+export default Ember.Component.extend({
+  classNames: ['controls'],
+
+  _version: 0,
+
+  _init: function() {
+    this.invalidate();
+  }.on('init'),
+
+  invalidate() {
+    this.set('_version', this.get('_version') + 1);
```

Comment:
I am not a huge fan of the `invalidate` method here.

I think a better approach would be to add a computed property for `Notification.permission`. Have your `isDeniedPermission`, `isDefaultPermission` etc depend on that new property.

Then instead of calling `invalidate` call `propertyDidChange` when you need to look at it.